components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -44,15 +46,42 @@ subprocess.check_call([
 
 // Configure the Python environments
 envs {
-    bootstrapDirectory = new File(buildDir, 'bootstrap')
-    envsDirectory = new File(buildDir, 'envs')
+    File gleanDirectory = new File(
+        project.getGradle().gradleUserHomeDir,
+        "glean"
+    )
+    bootstrapDirectory = new File(
+        gleanDirectory,
+        "bootstrap-${MINICONDA_VERSION}"
+    )
+    envsDirectory = new File(
+        gleanDirectory,
+        "envs-${MINICONDA_VERSION}"
+    )
     pipInstallOptions = "--trusted-host pypi.python.org --no-cache-dir"
 
     // Setup a miniconda environment. conda is used because it works
     // non-interactively on Windows, unlike the other options
-    conda "Miniconda3", "Miniconda3-4.5.11", "64", ["glean_parser==${GLEAN_PARSER_VERSION}"]
+    conda "Miniconda3", "Miniconda3-${MINICONDA_VERSION}", "64", ["glean_parser==${GLEAN_PARSER_VERSION}"]
 }
 
+// Setup the python environment before the build starts.
+task createBuildDir {
```

Comment:
Because the "fix" in `com.jetbrains.python.envs` only works if you're installing the build environment under build (`bootstrapDirectory` is under build).  If `bootstrapDirectory` points to another location, it still (temporarily) downloads the Miniconda installer to the `build` directory, which in this case now doesn't exist.  It's a pain, but only a minor one, given the advantages, IMHO.

Reply:
I see, makes sense!